code/game/machinery/doors/airlock.dm

```diff
@@ -1149,6 +1149,9 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
 					"<span class='notice'>You [welded ? "weld the airlock shut":"unweld the airlock"].</span>")
 				update_icon()
 		else if(obj_integrity < max_integrity)
+			if(icon_state != "closed")
```

Comment:
```suggestion
			if(AIRLOCK_OPEN)
```
I think this is how the airlock.dm file is checking for the airlock state.

Reply:
`AIRLOCK_OPEN` is a define for `3`, and used in `update_icon()` as the state parameter (or calculates) to set `icon_state` before passing to the overlay handler.

The actual state here is stored in `icon_state`, which is what we must check to determine current state. We also want to handle all cases where the door is **not** closed, rather than just open.

Reply:
That suggestion would always return the warning, as `3` is always true 😄

Reply:
Could you check the airlock's current state vs `AIRLOCK_OPEN`? I think that's what Henri meant

Reply:
AIRLOCK_OPEN, along with all the other state defines, are only used in `set_airlock_overlays()`, `do_animate()`, and `open()`, with the purpose of updating the icon via `update_icon()` as identifiers for what icon should be used during animation and post-animation. The actual state exists in `icon_state` as a string, and can only be one of the following:
* "open" (when opened)
* "closed" (when closed)
* "nonexistenticonstate" (when in the process of being opened, closed, emagged, or denying access)

Without an overhaul to airlock state code, the only place where state can be determined is two variables:
* `operating` (none, or one of the DOOR_X num defines if being operated)
* `icon_state` (as above)

**Short version:** Comparing to `AIRLOCK_OPEN` doesn't make sense, as that's used solely within update_icon calls.

Reply:
> Weird. Checking the icon state seems a bit scuffed, but if it's the easiest way to do it then it's probably fine

I agree. An alternative option is to compare layer, or we could look at the `density` boolean, as doors aren't interactable during transition anyway. This would be more compatible with sub-objects of airlock that do fancier work around animation.